### PR TITLE
feat: update basket data on all requests

### PR DESCRIPTION
**Pull request by abutterworth**
Also pass through get parameter to first getBasket request.

This will need a refactor in the future.

### CODE REVIEW

File: src/payment/coupon/data/sagas.js [hunk 1]
```diff
@@ -39,10 +32,13 @@ export function* handleRemoveCoupon(action) {
   yield put(removeCouponBegin());
   try {
     const result = yield call(deleteCoupon, action.payload.id);
-    yield put(fetchBasketSuccess(result));
+    yield call(handleBasketResult, result);
     yield put(removeCouponSuccess(result));
```
Comment (nasthagiri) [inline src/payment/coupon/data/sagas.js]: Does this remove a coupon success message?  Should that happen before handling the newly returned result?

File: src/payment/coupon/data/sagas.js [hunk 2]
```diff
@@ -18,18 +17,12 @@ export function* handleAddCoupon(action) {
   yield put(addCouponBegin());
   try {
     const result = yield call(postCoupon, action.payload.code);
-    yield put(fetchBasketSuccess(result));
+    yield call(handleBasketResult, result);
     yield call(handleMessages, result.messages, true);
```
Comment (nasthagiri) [inline src/payment/coupon/data/sagas.js]: Should `handleMessages` just be part of `handleBasketResult`?

File: src/payment/coupon/data/service.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import pick from 'lodash.pick';
 
-import { handleRequestError, applyConfiguration } from '../../../common/serviceUtils';
+import { applyConfiguration } from '../../../common/serviceUtils';
```
Comment (nasthagiri) [inline src/payment/coupon/data/service.js]: Is this import of `applyConfiguration` used?

File: src/payment/coupon/data/sagas.js (hunk 1 above)
Comment (nasthagiri) [inline src/payment/coupon/data/sagas.js]: Should removal of old messages just be part of `handleBasketResult`?

File: src/payment/data/service.js [hunk 4]
```diff
@@ -36,6 +39,26 @@ export function configureApiService(newConfig, newApiClient) {
     }
     return response;
   });
+
+  const decodeURLParams = (search) => {
```
Comment (nasthagiri) [inline src/payment/data/service.js]: Future thought:
I don't think it's necessary right now, but in future, we may want to forward other types of params as well.  For now, it seems UTM params aren't needed to be forwarded.

File: src/payment/data/sagas.js [hunk 5]
```diff
@@ -20,24 +20,32 @@ import { saga as cybersourceSaga } from '../cybersource';
 import { saga as payPalSaga } from '../paypal';
 import { handleErrors, handleMessages } from '../../feedback';
 
+export function* handleBasketResult(basketResult) {
```
Comment (nasthagiri) [inline src/payment/data/sagas.js]: Consider going through one more pass on DRY-ing further by including message displaying and error handling in here as well?

File: src/payment/coupon/data/sagas.js (hunk 2 above)
Comment (abutterworth) [reply src/payment/coupon/data/sagas.js]: We handle messages on success and errors on failure, but both need to populate the basket.

File: src/payment/coupon/data/service.js (hunk 3 above)
Comment (abutterworth) [reply src/payment/coupon/data/service.js]: Yes it's in there.

File: src/payment/data/sagas.js (hunk 5 above)
Comment (abutterworth) [reply src/payment/data/sagas.js]: There's some complication around how errors and messages are handled that make this less than straightforward. I'll address in a refactor later.

**Pull request merged**